spec/controllers/answers_controller_spec.rb

```diff
@@ -103,12 +79,16 @@
   describe 'DELETE #destroy' do
     it 'destroy answer' do
       answer
-      expect { delete :destroy, params: { question_id: question, id: answer } }.to change(Answer, :count).by(-1)
+      expect { delete :destroy, params: { question_id: question, id: answer } }.to change(question.answers, :count).by(-1)
```

Comment:
Here we should use `change(Answer, :count)`, now you're checking that the answer is _unregistered_ from the question.